pkg/azuredisk/nodeserver.go

```diff
@@ -216,6 +216,22 @@ func (d *Driver) NodePublishVolume(ctx context.Context, req *csi.NodePublishVolu
 	if err := d.mounter.Mount(source, target, "", mountOptions); err != nil {
 		return nil, status.Errorf(codes.Internal, "Could not mount %q at %q: %v", source, target, err)
 	}
+
+	// if resize is required, resize filesystem
+	if required, ok := req.GetVolumeContext()[resizeRequired]; ok && required == "true" {
```

Comment:
the mount happens first in NodeStageVolume, I think should move this code to that function, and there is no necessity to change code here if there is already resize in NodeStageVolume